core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalReader.java

```diff
@@ -130,7 +136,19 @@ public long getNextSequenceNumber() {
   }
 
   @Override
-  public Journal.JournalEntry read() throws IOException, InvalidJournalEntryException {
+  public InputStream getCheckpoint() {
+    return Preconditions.checkNotNull(mCheckpointStream, "mCheckpointStream");
+  }
+
+  @Override
+  public Journal.JournalEntry getEntry() {
+    Preconditions.checkState(mCheckpointStream == null,
+        "Should not call read() when a checkpoint is available");
```

Comment:
but we called `getEntry`, not `read`? Is it ever possible that `mNextEntry != null` and `mCheckpointStream != null`?

Reply:
oops, updated to `getEntry()` (method rename failure)

It's never possible for them both to be non-null. `advance()` is the only way to set them, it starts by setting them both to `null`, and it returns as soon as one of them is non-null